### PR TITLE
Update Package Dependencies

### PR DESCRIPTION
**Pull request by rdamico**
Update package to address detected security vulnerability in dependencies. It also contains fixes due to the package updates,

**Pre Updates:**
found 38585 vulnerabilities (38558 low, 17 moderate, 10 high)

**Post Updates:**
found 191 low severity vulnerabilities

**Changes**
- Updated packages

  | Package | From | To |
  |:--- |:---|:---|
  |aws-sdk|2.536.0|2.631.0|
  |@babel/core|^7.8.0|^7.9.0|
  |@babel/preset-env|^7.8.2|^7.9.5|
  |babel-core|^7.0.0-bridge.0|(latest - unchanged)|
  |babel-jest|^24.9.0|^25.3.0|
  |babel-loader|^8.0.6|^8.1.0|
  |eslint|^5.15.1|^6.8.0|
  |eslint-config-airbnb-base|^13.1.0|^14.1.0|
  |eslint-plugin-import|^2.18.2|^2.20.2|
  |husky|^4.0.7"|^4.2.5|
  |jest|^24.9.0|^25.3.0|
  |lint-staged|^9.5.0|^10.1.3|
  |serverless|^1.60.5|^1.67.3|
  |serverless-iam-roles-per-function|^2.0.2|(latest - unchanged)|
  |serverless-offline|^5.12.1|(Not changed due breaking changes in latest)|
  |serverless-plugin-stage-variables|^1.9.5|(latest - unchanged)|
  |serverless-webpack|^5.3.1| (latest - unchaged)|
  |webpack|^4.41.5| ^4.42.1|

### CODE REVIEW

**Comment by jamesallardice** [comment]
First comment - thanks for the exceptionally detailed commit messages! Will dig into this and review properly ASAP.

File: package.json [hunk 1]
```diff
@@ -14,26 +14,26 @@
     "node": "12.14.1"
   },
   "dependencies": {
-    "aws-sdk": "2.536.0"
+    "aws-sdk": "2.656.0"
```
Comment (jamesallardice) [inline package.json:17]: Minor one - we try to keep this in sync with the version that is bundled into the Node.js runtime in AWS Lambda (currently 2.631.0 - see [docs](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html)). That ensures we are using the same version locally - this dependency is effectively ignored and not included in the bundle we deploy to the Lambdas.

Comment (rdamico) [reply package.json]: Done

**Comment by jamesallardice** [comment]
I have been trying to dig into the named/default export issue and having just cleared out node_modules and started again I can no longer reproduce the problem. Could you give this another go and see if you still experience the issue please?

My output:

```
😭  serverless-starter git:(pkg-dep-updates) ❎  git revert 57ba18a20e3d5eec1aef45ae01f9cf76422932ed
[pkg-dep-updates ea7fc2d] Revert "Refactor to named exports only for handler"
 2 files changed, 7 insertions(+), 8 deletions(-)
😀  serverless-starter git:(pkg-dep-updates) ❎  npm start

> @orangejellyfish/serverless-starter@0.1.0 start /Users/jallardice/dev/orangejellyfish/serverless-starter
> serverless offline start

Serverless: WARNING: More than one matching handlers found for 'src/functions/hello/index'. Using 'src/functions/hello/index.js'.
Serverless: Bundling with Webpack...
Time: 1520ms
Built at: 04/16/2020 7:46:50 AM
                       Asset      Size  Chunks             Chunk Names
src/functions/hello/index.js  1.25 KiB       0  [emitted]  src/functions/hello/index
Entrypoint src/functions/hello/index = src/functions/hello/index.js
[0] ./src/functions/hello/index.js + 1 modules 1.02 KiB {0} [built]
    | ./src/functions/hello/index.js 580 bytes [built]
    | ./src/utils/run-warm/index.js 468 bytes [built]
Serverless: Watching for changes...
offline: Starting Offline: dev/us-east-1.
offline: Offline [http for lambda] listening on http://localhost:3002
Scheduling [hello] cron: [*/5 * * * *] input: undefined

   ┌─────────────────────────────────────────────────────────────────────────┐
   │                                                                         │
   │   GET | http://localhost:3000/dev/hello                                 │
   │   POST | http://localhost:3000/2015-03-31/functions/hello/invocations   │
   │                                                                         │
   └─────────────────────────────────────────────────────────────────────────┘

offline: [HTTP] server ready: http://localhost:3000 🚀
```

**Comment by rdamico** [comment]
So 
```bash
~ rm -rf node_modules
~ g revert 57ba18a
~ npm i
~ npm start

> @orangejellyfish/serverless-starter@0.1.0 start /home/rd/Development/orangejellyfish/serverless-starter
> serverless offline start

Serverless: WARNING: More than one matching handlers found for 'src/functions/hello/index'. Using 'src/functions/hello/index.js'.
Serverless: Bundling with Webpack...
Time: 1282ms
Built at: 04/16/2020 08:43:25
                       Asset      Size  Chunks             Chunk Names
src/functions/hello/index.js  1.25 KiB       0  [emitted]  src/functions/hello/index
Entrypoint src/functions/hello/index = src/functions/hello/index.js
[0] ./src/functions/hello/index.js + 1 modules 1.02 KiB {0} [built]
    | ./src/functions/hello/index.js 580 bytes [built]
    | ./src/utils/run-warm/index.js 468 bytes [built]
Serverless: Watching for changes...
offline: Starting Offline: dev/us-east-1.
offline: Offline [http for lambda] listening on http://localhost:3002
Scheduling [hello] cron: [*/5 * * * *] input: undefined

   ┌─────────────────────────────────────────────────────────────────────────┐
   │                                                                         │
   │   GET | http://localhost:3000/dev/hello                                 │
   │   POST | http://localhost:3000/2015-03-31/functions/hello/invocations   │
   │                                                                         │
   └─────────────────────────────────────────────────────────────────────────┘

offline: [HTTP] server ready: http://localhost:3000 🚀
offline: 
offline: Enter "rp" to replay the last request

offline: GET /dev/hello (λ: hello)
offline: Failure: offline: handler 'default' in /home/rd/Development/orangejellyfish/serverless-starter/.webpack/service/src/functions/hello/index is not a function
Error: offline: handler 'default' in /home/rd/Development/orangejellyfish/serverless-starter/.webpack/service/src/functions/hello/index is not a function
    at InProcessRunner.run (/home/rd/Development/orangejellyfish/serverless-starter/node_modules/serverless-offline/dist/lambda/handler-runner/in-process-runner/InProcessRunner.js:74:13)
    at processTicksAndRejections (internal/process/task_queues.js:94:5)
    at LambdaFunction.runHandler (/home/rd/Development/orangejellyfish/serverless-starter/node_modules/serverless-offline/dist/lambda/LambdaFunction.js:239:20)
    at hapiHandler (/home/rd/Development/orangejellyfish/serverless-starter/node_modules/serverless-offline/dist/events/http/HttpServer.js:457:18)
    at module.exports.internals.Manager.execute (/home/rd/Development/orangejellyfish/serverless-starter/node_modules/@hapi/hapi/lib/toolkit.js:45:28)
    at Object.internals.handler (/home/rd/Development/orangejellyfish/serverless-starter/node_modules/@hapi/hapi/lib/handler.js:46:20)
    at exports.execute (/home/rd/Development/orangejellyfish/serverless-starter/node_modules/@hapi/hapi/lib/handler.js:31:20)
    at Request._lifecycle (/home/rd/Development/orangejellyfish/serverless-starter/node_modules/@hapi/hapi/lib/request.js:312:32)
    at Request._execute (/home/rd/Development/orangejellyfish/serverless-starter/node_modules/@hapi/hapi/lib/request.js:221:9)
Failed to execute scheduled function: [hello] Error: Error: offline: handler 'default' in /home/rd/Development/orangejellyfish/serverless-starter/.webpack/service/src/functions/hello/index is not a function
```

**Comment by jamesallardice** [comment]
> Your example had no actual request made?

😂 that would explain why i couldn't reproduce it this morning... i blame the couple of beers i was drinking while looking into this night! sorry. back to the drawing board, i will do a bit more digging.

**Comment by rdamico** [comment]
InProcessRunner:run(....)  requires `src/functions/hello/index.js` and then passes it to `_interopRequireWildcard`. The attempts to the parse the required module.

```
function _interopRequireWildcard(obj) {
    if (obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache();
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {};
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var key in obj) {
        if (Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
```

Currently we fall out the bottom of this function having gone through for loop building a new object.

I did try looking at babel config to set `__esModule` but whatever i tried failed. You know a better approach?

**Comment by jamesallardice** [comment]
This is very strange because Serverless (via Webpack and the serverless-webpack plugin) appears to bundle the function correctly (you can find this in a zip file in `.serverless` after running `./node_modules/.bin/sls build`):

```
!function(e,t){for(var n in t)e[n]=t[n]}(exports,function(e){var t={};function n(r){if(t[r])return t[r].exports;var o=t[r]={i:r,l:!1,exports:{}};return e[r].call(o.exports,o,o.exports,n),o.l=!0,o.exports}return n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:r})},n.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},n.t=function(e,t){if(1&t&&(e=n(e)),8&t)return e;if(4&t&&"object"==typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:e}),2&t&&"string"!=typeof e)for(var o in e)n.d(r,o,function(t){return e[t]}.bind(null,o));return r},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,"a",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p="",n(n.s=0)}([function(e,t,n){"use strict";n.r(t),n.d(t,"hello",(function(){return o}));var r=e=>async(t,...n)=>"aws.events"===t.source?"pinged":e(t,...n);const o=async e=>({statusCode:200,body:JSON.stringify({message:"Go Serverless v1.0! Your function executed successfully!",input:e})});t.default=r(o)}]));
```

If you import the above into the Node REPL the object (and in particular the `default` property) is as expected:

`{ hello: [AsyncFunction: o], default: [AsyncFunction] }`

**Comment by rdamico** [comment]
Reverted serverless-offlne and reverted changs to index.js, index.spec.js and index.yml

**Comment by jamesallardice** [comment]
although might be worth updating your git config - your commits are not associated with your github account currently

**Pull request merged**